src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClient.cs

```diff
@@ -34,18 +36,26 @@ namespace Microsoft.Extensions.AI;
 /// invocation requests to that same function.
 /// </para>
 /// </remarks>
-public class FunctionInvokingChatClient : DelegatingChatClient
+public partial class FunctionInvokingChatClient : DelegatingChatClient
 {
+    /// <summary><see cref="ActivitySource"/> used for tracking function invocations.</summary>
+    private static readonly ActivitySource _activitySource = new(typeof(FunctionInvokingChatClient).FullName!);
+
+    /// <summary>The logger to use for logging information about function invocation.</summary>
```

Comment:
What would you do instead? This is the component that's invoking the functions and so it's the only one that has the data about it.

Reply:
I can't think of an alternative. Just wondering if we might want to follow a similar approach in other implementations for consistency.

Reply:
@SteveSandersonMS, opinions?

Reply:
Well technically we could add some other `UseFunctionCallLogging` middleware that people would independently put after `UseFunctionInvocation`. That would be able to log the function call/result contents, though it still wouldn't be able to have an activity around the actual function invocation itself, or to log exceptions from the function invocation. Also it wouldn't be obvious that people could/should put `UseFunctionCallLogging` in their pipeline.

So for logging at least, I feel pretty comfortable that the approach here is good. The consumer can pass in an `ILoggerFactory` if and only if they want it to be used.

For telemetry, though, I'm not as settled. As it stands it doesn't look like the consumer gets any way to control if or how this happens. It's a shame there's no `ITelemetryReceiver` or similar concept we could be passing in to make it controllable. I suppose we could pass in an `ActivitySource?` and use its presence to control this (and by default, `UseFunctionInvocation` would pass one in).

Maybe it would be sufficient just to have a `public bool EnableTelemetry { get; set; }` on `FunctionInvokingChatClient`, which either defaults to `true` or have `UseFunctionCallLogging` set it to true by default.

It's not that I have strong scenarios for needing *not* to log telemetry. Just perhaps in theory this may be used in scenarios where the app is definitely not collecting telemetry and all the work we're doing to format messages for it is pointless, or perhaps the app does collect unrelated telemetry but is not interested in function calling telemetry for some reason.